eng/codeOptimization.targets

```diff
@@ -1,18 +1,19 @@
 <Project>
 
-  <PropertyGroup Condition="'$(ApplyPartialNgenOptimization)' == ''">
-     <ApplyPartialNgenOptimization Condition="'$(IsReferenceAssembly)' == 'true'">false</ApplyPartialNgenOptimization>
-     <ApplyPartialNgenOptimization Condition="'$(GeneratePlatformNotSupportedAssembly)' == 'true' OR '$(GeneratePlatformNotSupportedAssemblyMessage)' != ''">false</ApplyPartialNgenOptimization>
+  <PropertyGroup Condition="'$(IsEligibleForNgenOptimization)' == ''">
+     <IsEligibleForNgenOptimization>true</IsEligibleForNgenOptimization>
+     <IsEligibleForNgenOptimization Condition="'$(IsReferenceAssembly)' == 'true'">false</IsEligibleForNgenOptimization>
+     <IsEligibleForNgenOptimization Condition="'$(GeneratePlatformNotSupportedAssembly)' == 'true' OR '$(GeneratePlatformNotSupportedAssemblyMessage)' != ''">false</IsEligibleForNgenOptimization>
```

Comment:
Something to consider here: today optimization data isn't represented by configuration, so we apply to same data to all builds: thus the need for these approximations of where not to apply the data.  I think it would be a better solution if the configuration information was preserved, or at least we were more intentional about the cases where we wanted to apply the same optimization data to multiple assemblies.

Reply:
Cc @brianrob on that.